spec/integration/worker_async_spec.rb

```diff
@@ -14,7 +14,7 @@ module Backend
       subject { Worker.new(async: true, job_fetcher: job_fetcher) }
 
       let(:storage) { Service.storage }
-      let(:resque_redis ) { job_fetcher.instance_variable_get(:@redis) }
+      let(:redis_client ) { storage.instance_variable_get(:@redis_async) }
```

Comment:
We have to use the client that `job_fetcher` uses. Otherwise, when the database for these is different, the test will not work. e.g.

```
CONFIG_REDIS_ASYNC=true CONFIG_INTERNAL_API_USER=system_app CONFIG_INTERNAL_API_PASSWORD=password CONFIG_QUEUES_MASTER_NAME=localhost:6379/5 CONFIG_REDIS_PROXY=localhost:6379/6 time -- bundle exec rake spec
```

I know that presently async client disregards logical database numbers. But
* we can sue separate databases
* I have in my backlog to add support for logical databases, see https://github.com/3scale/apisonator/pull/350#issuecomment-2020501532

Reply:
Makes sense, I made some changes. Ugly but effective